### PR TITLE
Do not skip full img tile classifier + Fix Sequencial Export Issue

### PR DESCRIPTION
**Pull request by eugene123tw**
### Summary

* Always predict full res image samples when the tile classifier is enabled as it may skip full-resolution images.
* Fix sequential model export issue by resetting partition count for tile classifier

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: otx/algorithms/detection/adapters/mmdet/utils/config_utils.py [hunk 1]
```diff
@@ -346,6 +346,10 @@ def patch_tiling(config, hparams, dataset=None):
             logger.info(f"Patch model from: {config.model.type} to CustomMaskRCNNTileOptimized")
             config.model.type = "CustomMaskRCNNTileOptimized"
 
+            for subset in ("val", "test"):
+                if config.data[subset].pipeline[0]["transforms"][-1]["type"] == "Collect":
+                    config.data[subset].pipeline[0]["transforms"][-1]["keys"].append("full_res_image")
+
```
Comment (jaegukhyun) [inline otx/algorithms/detection/adapters/mmdet/utils/config_utils.py:350]: Can we just add "full_res_image" to tile_pipeline.py directly?

**Review by jaegukhyun — COMMENTED** [review]
I left a comment

File: otx/algorithms/detection/adapters/mmdet/utils/config_utils.py (hunk 1 above)
Comment (eugene123tw) [reply otx/algorithms/detection/adapters/mmdet/utils/config_utils.py:350]: Then we will need to add it to every `simple_test` that supports tiling. I added this only when the tile classifier is enabled, so the change is only focused on this issue.

**Review by jaegukhyun — DISMISSED** [review]

File: otx/algorithms/common/adapters/mmdeploy/apis.py [hunk 2]
```diff
@@ -254,7 +254,7 @@ def extract_partition(
             model_name: str = "model",
         ):
             """Function for extracting partition."""
-
+            reset_mark_function_count()
```
Comment (sovrasov) [inline otx/algorithms/common/adapters/mmdeploy/apis.py:257]: Nice catch!

**Review by jaegukhyun — DISMISSED** [review]

**Comment by jaegukhyun** [comment]
Let's merge this PR after merging tiling optimization PR

**Review by sungmanc — DISMISSED** [review]

**Review by jaegukhyun — DISMISSED** [review]

**Review by sungmanc — DISMISSED** [review]

**Comment by eugene123tw** [comment]
image

**Review by jaegukhyun — APPROVED** [review]

File: otx/algorithms/detection/adapters/mmdet/models/detectors/custom_maskrcnn_tile_optimized.py [hunk 3]
```diff
@@ -171,12 +171,15 @@ def simple_test(self, img, img_metas, proposals=None, rescale=False):
...
+            full_res_image (bool, optional): if the image is full resolution or not. Defaults to False.
 
         Returns:
             tuple: MaskRCNN output
         """
-
         keep = self.tile_classifier.simple_test(img) > 0.45
+        if isinstance(full_res_image, bool):
+            full_res_image = [full_res_image]
+        keep = full_res_image[0] | keep
```
Comment (jaegukhyun) [inline otx/algorithms/detection/adapters/mmdet/models/detectors/custom_maskrcnn_tile_optimized.py:182]: Could you explain when full_res_image is not bool?

Comment (eugene123tw) [reply otx/algorithms/detection/adapters/mmdet/models/detectors/custom_maskrcnn_tile_optimized.py:182]: mmdetection collects `full_res_image` as list. Only NNCF collate it as a single boolean

**Review by goodsong81 — APPROVED** [review]

**Pull request merged**